### DISCUSSION TITLE
Is it possible to keep a service but return a specific value for a single method in a test

### DISCUSSION THREAD

**Discussion by handyCAPS** [help-me-mom/ng-mocks#8817] [Q&A] [answered]
I'm very sorry if this is a simple thing, but I can't get it to work and there doesn't seem to be any documentation on this. I have a component with a service that works fine with keep(), but I want to override one of its methods and return a specific value. I've tried to use keep() and then a mock(), but this does not work. 

```
beforeEach(() => {
    return MockBuilder(BoardComponent, SharedModule)
      .keep(WordService)
      // All other methods disappear after adding this
      .mock(WordService, {
        checkWordInList: () => true,
      });
  });
```

**Comment by satanTime** [comment, marked as answer]
Do not combine `.keep(WordService)` and `.mock(WordService, ...)`: mocking the same token replaces the kept service.

Keep the real service, get the exact instance used by the component, and replace only that method:

```ts
beforeEach(() =>
  MockBuilder(BoardComponent, SharedModule)
    .keep(WordService),
);

it('uses the requested result', () => {
  const fixture = MockRender(
    BoardComponent,
    undefined,
    { detectChanges: false },
  );

  const service = fixture.point.injector.get(WordService);
  ngMocks.stubMember(
    service,
    'checkWordInList',
    () => true,
  );

  fixture.detectChanges();

  // assertions
});
```

All other `WordService` methods remain real because `ngMocks.stubMember` changes only `checkWordInList`. The delayed change detection is useful when the method is called during an initial lifecycle hook; it can be omitted when the test calls the relevant behavior later.

Closing as answered.
